src/BreakfastSerializer/Serializer.php

```diff
@@ -63,6 +70,55 @@ public function getDepth()
     /**
      * @inheritdoc
      */
+    public function incrementCurrentDepth()
+    {
+        $this->currentDepth++;
+
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function decrementCurrentDepth()
+    {
+        $this->currentDepth =
+            (1 <= $this->currentDepth) ? 1 : $this->currentDepth = 1;
+
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getCurrentDepth()
+    {
+        return $this->currentDepth;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resetCurrentDepth()
+    {
+        $this->currentDepth = 1;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isWithinBounds()
+    {
+        $isValid = (self::MAX_DEPTH_NOT_SET === $this->maxDepth)
+            ? true : ($this->currentDepth <= $this->maxDepth)
+                ? true : false;
```

Comment:
real programmers write everything in nested ternaries

```c++
int i = 5;
string result = i % 2 == 0 ? "a" : i % 3 == 0 ? "b" : i % 5 == 0 ? "c" : i
% 7 == 0 ? "d" : "e";
```

On Fri, Apr 24, 2015 at 1:34 PM, Eddie Abou-Jaoude <notifications@github.com
> wrote:

> In src/BreakfastSerializer/Serializer.php
> <https://github.com/brettminnie/breakfast-serializer/pull/13#discussion_r29044499>
> :
>
> > +    /**
> > +     * @inheritdoc
> > +     */
> > +    public function resetCurrentDepth()
> > +    {
> > +        $this->currentDepth = 1;
> > +    }
> > +
> > +    /**
> > +     * @inheritdoc
> > +     */
> > +    public function isWithinBounds()
> > +    {
> > +        $isValid = (self::MAX_DEPTH_NOT_SET === $this->maxDepth)
> > +            ? true : ($this->currentDepth <= $this->maxDepth)
> > +                ? true : false;
>
> Haha...I am not even justifying that with a reply! [image: :wink:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brettminnie/breakfast-serializer/pull/13/files#r29044499>
> .
>